src/Http.re

```diff
@@ -23,10 +29,37 @@ external createServer : (clientT => unit) => serverT = "create_server";
 /* TODO: Make res different than req? */
 external requestOn : (clientT, (bool, bytes) => unit) => unit = "request_on";
 
+let parser = (req) => {
+  let fields = ref([]);
+  let values = ref([]);
+  let settings: HttpParser.http_parser_settings = {
+    on_message_begin:    _ => 0,
+    on_url:              (_, url, _) => {req.url = url; 0},
+    on_status:           _ => 0,
+    on_header_field:     (_, f, _) => { fields := fields^ @ [f]; 0},
+    on_header_value:     (_, v, _) => { values := values^ @ [v]; 0},
```

Comment:
if you accumulate these in reverse order it will be faster:
```
values := [v, ...values^]
```
O(1) in the length of the list vs O(n)
Not sure if the order of the headers matter, if they do you can reverse the whole list in the on_headers_complete.